testsuite/mockserver.py

```diff
@@ -14,21 +19,37 @@ class Mockserver:
     """
 
     def __init__(self, url):
-        self.client = ApyProxy(url, session=httpx.Client(verify=False, timeout=5))
+        self.client = ApyProxy(url, session=KuadrantClient(verify=False))
 
-    def _expectation(self, expectation_id, response_data):
+    def _expectation(self, expectation_id, json_data):
         """
-        Creates an Expectation with given response_data.
+        Creates an Expectation from given expectation json.
         Returns the absolute URL of the expectation
         """
-        json_data = {"id": expectation_id, "httpRequest": {"path": f"/{expectation_id}"}}
-        json_data.update(response_data)
+        json_data["id"] = expectation_id
+        json_data.setdefault("httpRequest", {})["path"] = f"/{expectation_id}"
 
         self.client.mockserver.expectation.put(json=json_data)
         # pylint: disable=protected-access
         return f"{self.client._url}/{expectation_id}"
 
-    def create_expectation(
+    def create_request_expectation(
+        self,
+        expectation_id,
+        headers: dict[str, list[str]],
+    ):
+        """Creates an Expectation - request with given headers"""
+        json_data = {
+            "httpRequest": {
+                "headers": headers,
+            },
+            "httpResponse": {
+                "body": "",
+            },
+        }
+        return self._expectation(expectation_id, json_data)
+
+    def create_response_expectation(
```

Comment:
What about keeping the original name and just adding `create_request_expectation`